### DISCUSSION TITLE
need help for filter or maybe different aproach for tables

### DISCUSSION THREAD

**Discussion by EdgarsLv** [supabase/supabase#6092] [Questions] [answered]
I need to get conversation id to get all messages for that conversation.

user_one: userId is logegIn user and user_two: id comes from params. Problem is that if i switch users then in code user_one will be id and user_two will be userId so i need to chek ---- user_one: userId && user_two: id || user_one: id && user_two: userId.
I don't get how to do that. Need help.

![image](https://user-images.githubusercontent.com/56837461/160256034-bf2150bb-ef80-4dbc-ac88-f44a4fa936ae.png)
conversations_id
![image](https://user-images.githubusercontent.com/56837461/160256399-9978e977-bd6d-4f98-8fb4-335537f52bc1.png)

and another problem if i switch users i would need to get user_one_fkey instead of user_two_fkey
![image](https://user-images.githubusercontent.com/56837461/160256553-6ece1f73-b955-4aa9-a037-9648d45e2a15.png)

Basically it's chat where i can click on registered user profile and start chat, if i start chat i'm user_one hes user_two. Problem is if he come to my profile and sends message then hes user_one and it creates another conversation because conversation where user_one is hes id doesn't exist, hes user_two in my conversation. So i need to chek both cases to verify if conversation already exist between these two persons.

Any help will be appreciated.

**Comment by EdgarsLv** [comment]
example. now loged in as laura,
![image](https://user-images.githubusercontent.com/56837461/160257036-11ba2734-f7b8-420e-8d2b-ab13b63c2e61.png)

now if i log in as osis, for conversation 1 user info will be osis not laura, because i need to fect foreign key with user_one here 
![image](https://user-images.githubusercontent.com/56837461/160257138-c11df822-4ab3-45ba-b43b-92a693212449.png)

**Comment by GaryAustin1** [comment]
Just sticking with the way you organized the table...
You can get the conversation with something like: (not tested exact string formating)
```
 .or(`user_one.eq.${userId},user_two.eq.${userId}`)  
``` 
Then get both userId's info with the foreign keys at at same time like this: https://supabase.com/docs/reference/javascript/select#query-the-same-foreign-table-multiple-times

Then in your display just pick the one that matches your userId's position one, or two for display...

**Comment by EdgarsLv** [comment]
i guess i don't really understand how to fetch multiple fkeys...

![image](https://user-images.githubusercontent.com/56837461/160274554-8c5b03e8-315d-467d-b22f-b31648643f68.png)

but still this doesn't solves problem, if i log in as oposite user then it's going to be -> user_one: userId, user_two: id. 
And even more i don't have 2nd id as argument at that place. I will have only logedIn user id. 

And .or() filter works only if i want to get all conversations for currently loged in user.

If i go to user profile and want to send message there and we already have conversation started, then i need to chek it like this --- user_one: userId && user_two: id || user_one: id && user_two: userId to keep chating in existing conversations.

to better understand;
There is going to page with all the chats and this filter works for that because finds all chats where im in.
![image](https://user-images.githubusercontent.com/56837461/160275359-62667224-3fc5-4409-a5d3-42a3deda83fb.png)

but it's possible to go directly to user profile and keep chatting there. to find correct chat i have to compare user_one: userId user_two: id for match. If other user goes directly to profile he won't find conversation because user_one and user_two doesn't match from hes side as id's are opposite.
![image](https://user-images.githubusercontent.com/56837461/160275342-644325b1-0f80-4d5c-be64-617b0473d06c.png)

**Comment by GaryAustin1** [comment, marked as answer]
Oh yeah,
Should have been for just 1 conversation:
```
 .or(`and(user_one.eq.${userId}, user_two.eq.${id}), and(user_two.eq.${userId}, user_one.eq.${id})`)  
```

https://postgrest.org/en/stable/api.html#logical-operators shows form inside the .or

**Reply by EdgarsLv** [reply]
thx for this

**Comment by EdgarsLv** [comment]
can't use twice.
![image](https://user-images.githubusercontent.com/56837461/160291189-e62b855a-1c19-4f6e-92dc-e2dbc4b14574.png)

**Reply by GaryAustin1** [reply]
Use two different alias, like user1:user_one , user2:user_two.  The first part is an alias, not a table name, the system figures out the table from the foreign key itself.

In your code you will need to see if user_one in the data object matches userId, if so you use user1 info, if not user user2 info.

**Reply by EdgarsLv** [reply]
ok this works. thank you for help!
